warden/group/handler.go

```diff
@@ -66,24 +77,65 @@ func (h *Handler) SetRoutes(r *httprouter.Router) {
 //       oauth2: hydra.groups
 //
 //     Responses:
-//       200: findGroupsByMemberResponse
+//       200: findGroupsResponse
 //       401: genericError
 //       403: genericError
 //       500: genericError
 func (h *Handler) FindGroupNames(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 	var ctx = r.Context()
 	var member = r.URL.Query().Get("member")
 
-	g, err := h.Manager.FindGroupNames(member)
-	if err != nil {
+	accessReq := &firewall.TokenAccessRequest{
+		Resource: GroupsResource,
+		Action:   "get",
+	}
+
+	if member != "" {
+		accessReq.Resource = fmt.Sprintf(GroupResource, member)
+	}
+
+	if _, err := h.W.TokenAllowed(ctx, h.W.TokenFromRequest(r), accessReq, Scope); err != nil {
 		h.H.WriteError(w, r, err)
 		return
 	}
 
-	if _, err := h.W.TokenAllowed(ctx, h.W.TokenFromRequest(r), &firewall.TokenAccessRequest{
-		Resource: fmt.Sprintf(GroupResource, member),
-		Action:   "get",
-	}, Scope); err != nil {
+	if member != "" {
+		g, err := h.Manager.FindGroupNames(member)
+
+		if err != nil {
+			h.H.WriteError(w, r, err)
+			return
+		}
+
+		h.H.Write(w, r, g)
+		return
+	}
+
+	val := r.URL.Query().Get("offset")
```

Comment:
Can we wrap this and the parsing in it's own function? It would allow for better readability and a shorter function body, e.g.

func intFromQuery(r *http.Request, key string) (int, error)

Reply:
Sure, I didn't see many (any?) separate functions for these things, so I was just trying to maintain the existing style.

Reply:
No worries, most of them are encapsulated in their own libraries (e.g. the whole json/statuscode/error stuff) and thus a kind of invisible to the naked eye :)